forum/users/serializers.py

```diff
@@ -56,5 +61,19 @@ def create(self, validated_data):
         user = User.objects.create_user(**validated_data)
         if roles_data:
             user.roles.set(roles_data)
+            
+        token = RefreshToken.for_user(user).access_token
+        
+        
+        activation_url = f"http://localhost:8000{reverse('activate', kwargs={'token': str(token)})}"
+        
+        
+        send_mail(
+            subject='Activate Your Account',
+            message=f'Hi {user.first_name},\n\nPlease click the link below to activate your account:\n{activation_url}',
+            from_email=settings.EMAIL_HOST_USER,  
+            recipient_list=[user.email],
+            fail_silently=False,
+        )
```

Comment:
If the user already exists, or if sending the email fails, you should handle these cases explicitly.
Wrap the email sending in a `try-except` block to handle potential issues gracefully.